app/scripts/controllers/network/network-controller.test.ts

```diff
@@ -250,360 +119,33 @@ const BLOCKED_INFURA_RESPONSE = {
-  }
-}
-
-/**
- * Handles mocking requests made by NetworkController for an Infura network.
+ * A response object for a request that has been geoblocked by Infura.
  */
-class InfuraNetworkCommunications extends NetworkCommunications<InfuraNetworkCommunicationsOptions> {
-  /**
-   * Constructs an InfuraNetworkCommunications.
-   *
-   * @param args - The arguments.
-   * @param args.infuraProjectId - TODO.
-   * @param args.infuraNetwork - TODO.
-   */
-  constructor({
-    infuraProjectId = DEFAULT_INFURA_PROJECT_ID,
-    infuraNetwork,
-  }: InfuraNetworkCommunicationsOptions) {
-    super({
-      options: { infuraProjectId, infuraNetwork },
-      requestBaseUrl: `https://${infuraNetwork}.infura.io`,
-      requestPath: `/v3/${infuraProjectId}`,
-    });
-  }
-
-  /**
-   * Constructs a new InfuraNetworkCommunications object using a different set
-   * of options, using the options from this instance as a base.
-   *
-   * @param overrides - Options with which you want to extend the new
-   * InfuraNetworkCommunications.
-   */
-  with(
-    overrides: Partial<InfuraNetworkCommunicationsOptions> = {},
-  ): InfuraNetworkCommunications {
-    return new InfuraNetworkCommunications({
-      ...this.options,
-      ...overrides,
-    });
-  }
-
-  protected getDefaultNumTimesToMockEthBlockNumber(): number {
-    return 1;
-  }
-}
+const BLOCKED_INFURA_JSON_RPC_ERROR = ethErrors.rpc.internal(
+  JSON.stringify({ error: 'countryBlocked' }),
+);
 
 /**
- * Handles mocking requests made by NetworkController for a non-Infura network.
+ * A response object for a unsuccessful request to any RPC method. It is assumed
+ * that the error here is insignificant to the test.
  */
-class CustomNetworkCommunications extends NetworkCommunications<CustomNetworkCommunicationsOptions> {
-  /**
-   * Constructs a CustomNetworkCommunications.
-   *
-   * @param args - The arguments.
-   * @param args.customRpcUrl - The URL that points to the RPC endpoint.
-   */
-  constructor({ customRpcUrl }: CustomNetworkCommunicationsOptions) {
-    super({
-      options: { customRpcUrl },
-      requestBaseUrl: customRpcUrl,
-      requestPath: '/',
-    });
-  }
-
-  /**
-   * Constructs a new CustomNetworkCommunications object using a different set
-   * of options, using the options from this instance as a base.
-   *
-   * @param overrides - Options with which you want to extend the new
-   * CustomNetworkCommunications.
-   */
-  with(
-    overrides: Partial<CustomNetworkCommunicationsOptions> = {},
-  ): CustomNetworkCommunications {
-    return new CustomNetworkCommunications({
-      ...this.options,
-      ...overrides,
-    });
-  }
-}
+const GENERIC_JSON_RPC_ERROR = ethErrors.rpc.internal(
+  JSON.stringify({ error: 'oops' }),
+);
 
 describe('NetworkController', () => {
-  let clock: SinonFakeTimers;
-
   beforeEach(() => {
     // Disable all requests, even those to localhost
     nock.disableNetConnect();
-    // Faking timers ends up doing two things:
```

Comment:
We don't need to use fake timers because the block tracker never gets started and there is no retry logic because the middleware never gets used. So that simplifies things a bit.